src/acl/Gadgets.cc

```diff
@@ -298,17 +289,10 @@ aclDestroyDenyInfoList(AclDenyInfoList ** list)
 {
     AclDenyInfoList *a = NULL;
     AclDenyInfoList *a_next = NULL;
-    AclNameList *l = NULL;
-    AclNameList *l_next = NULL;
 
     debugs(28, 8, "aclDestroyDenyInfoList: invoked");
 
     for (a = *list; a; a = a_next) {
```

Comment:
Can we just `delete *list` here? If not, please add a source code comment to explain why not. If we can, then this will fix the current err_page_name leak AFAICT (err_page_name is freed in the destructor but not here). If you cannot tell, please add a `TODO: delete *list if possible` comment.

Reply:
This PR is just replacing the acl_list. The AclDenyInfoList::next structure needs replacing with another std:: container type, which will make this whole function irrelevant and go away.

Also the loop over *list is the pattern for efficient unknown-size tree deletion without overflowing the stack. The "delete a" a few lines below invokes the relevant destructor for releasing err_page_name.

Reply:
Sorry, I misinterpreted these loops. FTR, the answer to my original question is "The PR code already deletes `*list` (essentially, via `delete a`), but we still need to iterate over lists to delete each of them, and that is what the remaining/outer loop does". This change request is addressed.